core/domain/exp_domain.py

```diff
@@ -3162,7 +3279,11 @@ def _choices_should_be_unique_and_non_empty(
                         invalid_rules.append(rule_spec)
             for invalid_rule in invalid_rules:
                 answer_group['rule_specs'].remove(invalid_rule)
-            # Removal of answer group if it is valid check.
+            if len(answer_group['rule_specs']) == 0:
+                empty_ans_groups.append(answer_group)
+
+        for empty_ans_group in empty_ans_groups:
+            answer_groups.remove(empty_ans_group)
```

Comment:
should this all be in an else statement?

Reply:
Yep, because we need this after all the rules have been checked.

Reply:
I think this would not go inside the else part, can you please guide me here why it should be in else part?
Thanks